### PR TITLE
feat: [CDS-49884]: TAS RollingDeployment and RollingRollback step

### PR DESCRIPTION
**Pull request by PratyushGar**
#### Screenshots
image
image
image
image
image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/TasRollingDeploymentStep/TasRollingDeploymentStep.tsx [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+                  label={getString('pipelineSteps.timeoutLabel')}
+                  multiTypeDurationProps={{
+                    enableConfigureOptions: false,
+                    expressions,
+                    disabled: readonly,
+                    allowableTypes
+                  }}
+                />
+                {getMultiTypeFromValue(values.timeout) === MultiTypeInputType.RUNTIME && (
+                  <ConfigureOptions
```
Comment (vkbansal-harness) [inline src/modules/75-cd/components/PipelineSteps/TasRollingDeploymentStep/TasRollingDeploymentStep.tsx:128]: Please use inbuilt configuration options

**Review by vkbansal-harness — CHANGES_REQUESTED** [review]

File: src/modules/75-cd/components/PipelineSteps/TasRollingDeploymentStep/__tests__/__snapshots__/TasRollingDeployment.test.tsx.snap [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`TASRollingDeploymentStepStep tests Variables view renders fine 1`] = `
+<div>
+  <div
+    class="variablesListTable variablePaddingL3"
+  />
+</div>
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/TasRollingDeploymentStep/__tests__/__snapshots__/TasRollingDeployment.test.tsx.snap:8]: snapshot only has container divs

File: src/modules/75-cd/components/PipelineSteps/TasRollingRollbackStep/__tests__/__snapshots__/TasRollingRollbackStep.test.tsx.snap [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`TASRollingRollbackStepStep tests Variables view renders fine 1`] = `<div />`;
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/TasRollingRollbackStep/__tests__/__snapshots__/TasRollingRollbackStep.test.tsx.snap:3]: Can you check these snapshots again?

File: src/modules/75-cd/components/PipelineSteps/TasRollingDeploymentStep/TasRollingDeploymentStep.tsx (hunk 1 above)
Comment (PratyushGar) [reply src/modules/75-cd/components/PipelineSteps/TasRollingDeploymentStep/TasRollingDeploymentStep.tsx]: Done 👍

File: src/modules/75-cd/components/PipelineSteps/TasRollingDeploymentStep/__tests__/__snapshots__/TasRollingDeployment.test.tsx.snap (hunk 2 above)
Comment (PratyushGar) [reply src/modules/75-cd/components/PipelineSteps/TasRollingDeploymentStep/__tests__/__snapshots__/TasRollingDeployment.test.tsx.snap]: Thanks, updated.

File: src/modules/75-cd/components/PipelineSteps/TasRollingRollbackStep/__tests__/__snapshots__/TasRollingRollbackStep.test.tsx.snap (hunk 3 above)
Comment (PratyushGar) [reply src/modules/75-cd/components/PipelineSteps/TasRollingRollbackStep/__tests__/__snapshots__/TasRollingRollbackStep.test.tsx.snap:3]: Yes, empty snapshot is desired, as rollback steps have no spec, so empty div is generated, as generic (name, identifier, timeout) field is handled globally.

**Review by rameshx — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**